l1-contracts/contracts/upgrades/GatewayUpgrade.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.24;
+
+import {Diamond} from "../state-transition/libraries/Diamond.sol";
+import {BaseZkSyncUpgrade, ProposedUpgrade} from "./BaseZkSyncUpgrade.sol";
+
+import {DataEncoding} from "../common/libraries/DataEncoding.sol";
+
+import {AdminFacet} from "../state-transition/chain-deps/facets/Admin.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @notice This upgrade will be used to migrate Era to be part of the hyperchain ecosystem contracts.
+contract GatewayUpgrade is BaseZkSyncUpgrade, AdminFacet {
+    /// @notice The main function that will be called by the upgrade proxy.
+    /// @param _proposedUpgrade The upgrade to be executed.
+    function upgrade(ProposedUpgrade calldata _proposedUpgrade) public override returns (bytes32) {
+        (
+            address l1DAValidator,
+            address l2DAValidator,
```

Comment:
those will be different for different chains as some way want to have validium ones. I would suggest to include those not as part of the upgrade file, but just require that `ChainAdmin` executes those in a multicall fasion.